qa/rgw/store/sfs/tests/test-sfs-object-locking.py

```diff
@@ -0,0 +1,293 @@
+                                                       Key="key.1",
+                                                       VersionId=self.ObjVersions["key.1.1"])
+
+        self.check_object_retention(response, "GOVERNANCE", 7, "Days")
+
+    def test_object_locking_object_compliance(self):
+        self.ensure_bucket(ObjectLockingTests.BUCKET_NAME_1, True)
+
+        retainUntilDate = datetime.utcnow() + timedelta(days=6)
+
+        self.s3_client.put_object(Bucket=ObjectLockingTests.BUCKET_NAME_1,
+                                  Key="key.1",
+                                  Body="data.3",
+                                  ObjectLockMode="COMPLIANCE",
+                                  ObjectLockRetainUntilDate=retainUntilDate)
+
+        response = self.s3_client.list_object_versions(Bucket=ObjectLockingTests.BUCKET_NAME_1,
+                                                       Prefix="key.1")
+
+        for version in response["Versions"]:
+          if version["Key"] == "key.1" and version["IsLatest"] == True:
+            self.ObjVersions["key.1.3"] = version["VersionId"]
+            print(self.ObjVersions["key.1.3"])
+
+        response = self.s3_client.get_object_retention(Bucket=ObjectLockingTests.BUCKET_NAME_1,
+                                                       Key="key.1",
+                                                       VersionId=self.ObjVersions["key.1.3"])
+
+        self.check_object_retention(response, "COMPLIANCE", 6, "Days")
+
+        try:
+          response = self.s3_client.delete_object(Bucket=ObjectLockingTests.BUCKET_NAME_1,
+                                                  Key="key.1",
+                                                  VersionId=self.ObjVersions["key.1.3"])
+        except botocore.exceptions.ClientError as error:
+           self.assertTrue(error.response["ResponseMetadata"]["HTTPStatusCode"] == 403)
+           self.assertTrue(error.response["Error"]["Code"] == "AccessDenied")
+           self.assertTrue(error.response["Error"]["Message"] == "forbidden by object lock")
+
+        # test deletion is allowed after retainUntilDate expires
+
+        retainUntilDate = datetime.utcnow() + timedelta(seconds=1)
+
+        self.s3_client.put_object(Bucket=ObjectLockingTests.BUCKET_NAME_1,
+                                  Key="key.1",
+                                  Body="data.3.1",
+                                  ObjectLockMode="COMPLIANCE",
+                                  ObjectLockRetainUntilDate=retainUntilDate)
+
+        response = self.s3_client.list_object_versions(Bucket=ObjectLockingTests.BUCKET_NAME_1,
+                                                       Prefix="key.1")
+
+        for version in response["Versions"]:
+          if version["Key"] == "key.1" and version["IsLatest"] == True:
+            self.ObjVersions["key.1.3.1"] = version["VersionId"]
+            print(self.ObjVersions["key.1.3.1"])
+
+        time.sleep(2)
```

Comment:
Consider removing or adding a comment on why the tests have to wait 2 seconds between list versions and delete. (here and further down)